### ISSUE TITLE
Request: Blenderplayer headless mode

### ISSUE THREAD

**Issue by organicpencil** [UPBGE/blender#12]
For running server-side game simulations.  Ideally a command-line switch, but I'd also settle for a build option.

**Comment by lordloki** [comment]
This feature will be available when Benoit merges with decklink branch.
Using bge.logic.setRender(False) will be enough to deactive the render pass

**Comment by BluePrintRandom** [comment]
so it also turns off culling etc?

**Comment by panzergame** [comment]
@BluePrintRandom : i think not since armature animations are culling dependent.

**Comment by panzergame** [comment]
@lordloki: Any new of the benoit's branch ?

**Comment by lordloki** [comment]
I think that benoit said that he will commit it in phase 1-2 of this cycle before 2.78 release but I don't have any additional info.

**Comment by DCubix** [comment]
Can we reconsider this? @panzergame

**Comment by youle31** [comment]
@dcubix: It is implemented in ge videotexture. Commit from benoit bolsee and dfelinto

**Comment by panzergame** [comment]
To be clear @pqftgs, a headless mode means no render but window or simply no window ?
Anyway if your in linux you can setup a xvfb server. It run pretty fine, i use it for the buildbot to access to the build version.

**Comment by pqftgs** [comment]
@panzergame: Headless would mean no window / X server not required.

XVFB + disable render looks like it will work, despite the unnecessary overhead.  This workaround is good enough for my needs.

True headless support would be better, but it's not a huge priority.

**Comment by DCubix** [comment]
Added a "high priority" label. This request is really old

**Comment by pqftgs** [comment]
To do this properly we also need to disable input and sound.  There's already WITH_HEADLESS that disables the window, just need ifdefs and stubs to make the game engine work.  Things are kinda hectic right now but it's on my to-do list.

**Comment by lordloki** [comment]
For sound exists WITH_AUDASPACE.

**Comment by panzergame** [comment]
@pqftgs : We can avoid creating a windows but we can't do anything without an opengl context, because it will assume to comment every lines calling something related to OpenGL (including bgl) with the WITH_HEADLESS macro.

**Comment by pqftgs** [comment]
@panzergame that's what stubs are for.
Again, it could be awhile (or never) before I work on this.  Life is full of uncertainty.  Can probably close if you want the issue tracker to look clean.  Already have a semi-viable workaround.

**Comment by atoms118** [comment]
Is there a reason why this was closed?

**Comment by BluePrintRandom** [comment]
Perhaps we should re-open this?

**Comment by youle31** [comment]
It was closed to clean up the tracker and focus on priorities I guess. All depends on benevolent developers motivation.

**Comment by atoms118** [comment]
ELI5, is there a current status on this? My understanding is you can already disable rendering, that the actual issue is it still depends on X11 just to create a window, or is there something more going on here?
